db/python/layers/cohort.py

```diff
@@ -249,3 +250,23 @@ async def create_cohort_from_criteria(
             description=description,
             template_id=template_id,
         )
+
+    async def update_cohort(self, cohort_update_body: CohortUpdateBody, cohort_id: int):
+        """update Cohort given id"""
+        cohorts, project_ids = await self.ct.query(
+            CohortFilter(id=GenericFilter(eq=cohort_id))
+        )
+
+        self.connection.check_access_to_projects_for_ids(
+            project_ids, allowed_roles=FullWriteAccessRoles
+        )
+
+        if not cohorts:
```

Comment:
Somewhere in here we'll need some handling to raise an error if you are trying to mark a cohort as ACTIVE when it is invalid due to inactive sgs/samples

Reply:
Addressed in https://github.com/populationgenomics/metamist/commit/e9279f7969fee6cb943518805ee726c3ece2ec80